src/Implementation.fs

```diff
@@ -0,0 +1,95 @@
+module Async_rpc.Implementation
+
+open Core_kernel
+open Async_rpc.Protocol
+open Bin_prot.Common
+open Async_rpc
+open Transport
+
+type 'connection_state t =
+    { rpc_tag: Rpc_tag.t
+      rpc_version: int64
+      partial_fun: 'connection_state
+          -> Bin_prot.Nat0.t Query.t
+          -> buf
+          -> pos_ref
+          -> Transport.Writer.t
+          -> Result.t<unit, Rpc_error.t>
+<<<<<<< HEAD
+=======
+    (*; implement: ('connection_state -> Buffer.t -> Buffer.t)*)
+>>>>>>> ec23e7a927aa331442e8fa83058b083345c54c1d
+     }
+
+let real_function
```

Comment:
Instead of [real_function] we should give it a name that explains what is happening